frozen-abi/src/abi_example.rs

```diff
@@ -137,25 +137,12 @@ tuple_example_impls! {
     }
 }
 
-// Source: https://github.com/rust-lang/rust/blob/ba18875557aabffe386a2534a1aa6118efb6ab88/src/libcore/array/mod.rs#L417
-macro_rules! array_example_impls {
-    {$n:expr, $t:ident $($ts:ident)*} => {
-        impl<T> AbiExample for [T; $n] where T: AbiExample {
-            fn example() -> Self {
-                [$t::example(), $($ts::example()),*]
-            }
-        }
-        array_example_impls!{($n - 1), $($ts)*}
-    };
-    {$n:expr,} => {
-        impl<T> AbiExample for [T; $n] {
-        fn example() -> Self { [] }
-        }
-    };
+impl<const N: usize, T: AbiExample + Copy> AbiExample for [T; N] {
+    fn example() -> Self {
+        [T::example(); N]
+    }
 }
```

Comment:
elegance of const generitics...

offtopic: really wish Default is reimplementd like this as soon as possible...